### PR TITLE
Add submitReplacement(target, Optional<T> newValue) and submitConditi…

### PR DESCRIPTION
**Pull request by gradycsjohnson**
…onalReplacement(target, Optional<T> newValue, etc)  and make them default

Implementation of #22. I made the Optional implementations default and had them call the original methods. 

If we like this approach I will make `requiredValue` optional and have it call `submitInitialization()`

### CODE REVIEW

File: bosk-testing/src/main/java/io/vena/bosk/drivers/DriverConformanceTest.java [hunk 1]
```diff
@@ -242,7 +243,7 @@ void testOptional() throws InvalidTypeException {
 		driver.submitReplacement(ref, TestValues.blank().withString("changed"));
 		assertCorrectBoskContents();
 
-		assertThrows(NullPointerException.class, ()->driver.submitReplacement(ref, null));
+		assertThrows(NullPointerException.class, ()->driver.submitReplacement(ref, Optional.of(null)));
```
Comment (prdoyle) [inline bosk-testing/src/main/java/io/vena/bosk/drivers/DriverConformanceTest.java:246]: This is no longer testing the same thing. `Optional.of` throws `NullPointerException` itself, so this isn't testing the bosk library anymore.

If you're getting an error from the compiler, you might want to replace `null` with `(TestValues)null`.

**Review by prdoyle — CHANGES_REQUESTED** [review]
A request for a change in the unit test...

File: bosk-testing/src/main/java/io/vena/bosk/drivers/DriverConformanceTest.java (hunk 1 above)
Comment (gradycsjohnson) [reply bosk-testing/src/main/java/io/vena/bosk/drivers/DriverConformanceTest.java:246]: Hm. This sitll doesn't seem to satisfy the compiler...

**Comment by prdoyle** [comment]
Superseded by #45

**Pull request closed without merging**